### PR TITLE
Mark `finish_reason` and other properties as required

### PR DESCRIPTION
**Pull request by rattrayalex**
Attempts to resolve https://github.com/openai/openai-node/issues/82

Note I am very uncertain whether these changes are correct; they are just guesses. Please audit them before merging.

### CODE REVIEW

File: openapi.yaml [hunk 1]
```diff
@@ -2155,6 +2159,8 @@ components:
           type: array
           items:
             type: object
+            required:
+              - index
```
Comment (rattrayalex) [inline openapi.yaml:2163]: Note that I assume that `index` would always be present, but `delta` and `finish_reason` may not be here; I have not checked whether that tends to be the case.

File: openapi.yaml [hunk 2]
```diff
@@ -2233,6 +2239,11 @@ components:
           type: array
           items:
             type: object
+            required:
+              - text
+              - index
+              - logprobs
```
Comment (rattrayalex) [inline openapi.yaml:2245]: I am not sure if the `logprobs` key would always be present; it is marked as nullable.

**Review by rattrayalex — COMMENTED** [review]
Noting a few things I was particularly unsure of.

**Review by rattrayalex — COMMENTED** [review]
Noting a few things I was particularly unsure of.

File: openapi.yaml (hunk 1 above)
Comment (athyuttamre) [reply openapi.yaml:2163]: `delta` and `finish_reason` are also always present, we can make this required.

File: openapi.yaml (hunk 2 above)
Comment (athyuttamre) [reply openapi.yaml:2245]: This is a bug — the `v1/edits` response doesn't support `logprobs` as a parameter or a response field. Can we remove from both request and response?

**Review by athyuttamre — COMMENTED** [review]
Couple of fixes, thanks for doing this!

File: openapi.yaml (hunk 1 above)
Comment (rattrayalex) [reply openapi.yaml:2163]: Ah, then presumably `finish_reason` should be marked as nullable?

Comment (rattrayalex) [reply openapi.yaml:2163]: And is `delta` nullable, or would it be `''` on the last chunk when it's just a finish reason?

**Comment by rattrayalex** [comment]
Thanks Atty! Back to you – want to make sure we get the typing around `delta` right.

File: openapi.yaml (hunk 1 above)
Comment (athyuttamre) [reply openapi.yaml:2163]: finish_reason is nullable, whereas delta is always a dictionary with every key (and sun key) optional.

**Review by athyuttamre — COMMENTED** [review]
Clarified about delta / finish_reason

**Pull request merged**